src/colorizer/utils/react_utils.ts

```diff
@@ -510,6 +515,17 @@ export const useAnnotations = (): AnnotationState => {
     setIsAnnotationEnabled(false);
   };
 
+  const replaceAnnotationData = (newAnnotationData: AnnotationData): void => {
+    annotationDataRef.current = newAnnotationData;
+    if (newAnnotationData.getLabels().length > 0) {
+      if (currentLabelIdx === null) {
+        setCurrentLabelIdx(0);   
+      } else if (currentLabelIdx >= newAnnotationData.getLabels().length) {
```

Comment:
nit: instead of calling getLabels() twice, might be more efficient to store length in a const.